File: Java/Medium/WrongAnswer/coordinate-with-maximum-network-quality_4.java
class Solution {
    public int[] bestCoordinate(int[][] towers, int radius) {
        
        int[] result = new int[2];
        int max = 0;
        for (int i = 0; i < towers.length; i++) {
            int quality = 0;
            for (int j = 0; j < towers.length; j++) {
                int distance = (int) Math.sqrt(Math.pow(towers[i][0] - towers[j][0], 2) + Math.pow(towers[i][1] - towers[j][1], 2));
                if (distance <= radius) {
                    quality += (int) Math.floor(towers[j][2] / (1.0 + distance));
                }
            }
            if (quality > max) {
                max = quality;
                result[0] = towers[i][0];
                result[1] = towers[i][1];
            } else if (quality == max) {
                if (towers[i][0] < result[0]) {
                    result[0] = towers[i][0];
                    result[1] = towers[i][1];
                } else if (towers[i][0] == result[0] && towers[i][1] < result[1]) {
                    result[1] = towers[i][1];
                }
            }
        }
        return result;
    }
}

File: C++/Medium/Accepted/coordinate-with-maximum-network-quality_4.cpp
class Solution {
public:
    vector<int> bestCoordinate(vector<vector<int>>& towers, int radius) {
        
        vector<int> res{0, 0};
        int ans = 0;
        for (int x = 0; x <= 50; x++) {
            for (int y = 0; y <= 50; y++) {
                int sum = 0;
                for (auto& t : towers) {
                    int d = (x - t[0]) * (x - t[0]) + (y - t[1]) * (y - t[1]);
                    if (d <= radius * radius) {
                        sum += floor(t[2] / (1.0 + sqrt(d)));
                    }
                }
                if (sum > ans) {
                    ans = sum;
                    res = {x, y};
                }
            }
        }
        return res;
    }
};